### PR TITLE
feat(rtc_manager_panel): add rtc all approval

### PR DESCRIPTION
**Pull request by taikitanaka3**
Signed-off-by: taikitanaka3 <taiki.tanaka@tier4.jp>

## Description

- add rtc exe mode and wait mode button

https://user-images.githubusercontent.com/65527974/193736296-f37a2bf7-d20b-4b61-9a61-11024817806f.mp4

## Related links

resolve https://github.com/autowarefoundation/autoware.universe/pull/2009

 '[Tier4 INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-21754)'

## Tests performed

not yet

## Notes for reviewers

this can easily tested by psim

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by taikitanaka3** [comment]
@rej55 
Can you review this PR please?

File: common/rtc_manager_rviz_plugin/src/rtc_manager_panel.cpp [hunk 1]
```diff
@@ -171,6 +136,20 @@ RTCManagerPanel::RTCManagerPanel(QWidget * parent) : rviz_common::Panel(parent)
   }
   v_layout->addWidget(auto_mode_table_);
 
+  // execution
+  auto * rtc_exe_layout = new QHBoxLayout;
+  {
+    exec_button_ptr_ = new QPushButton("exe mode");
```
Comment (rej55) [inline common/rtc_manager_rviz_plugin/src/rtc_manager_panel.cpp:142]: I think it is better not to abbreviate.
How about following suggestion?

```suggestion
    exec_button_ptr_ = new QPushButton("Execute");
```

File: common/rtc_manager_rviz_plugin/src/rtc_manager_panel.cpp [hunk 2]
```diff
@@ -171,6 +136,20 @@ RTCManagerPanel::RTCManagerPanel(QWidget * parent) : rviz_common::Panel(parent)
...
+    exec_button_ptr_->setCheckable(false);
+    connect(exec_button_ptr_, &QPushButton::clicked, this, &RTCManagerPanel::onClickExecution);
+    rtc_exe_layout->addWidget(exec_button_ptr_);
+    wait_button_ptr_ = new QPushButton("wait mode");
```
Comment (rej55) [inline common/rtc_manager_rviz_plugin/src/rtc_manager_panel.cpp:146]:
```suggestion
    wait_button_ptr_ = new QPushButton("Wait");
```

**Review by rej55 — COMMENTED** [review]
Thank you!
I add some suggestions.
Could you check them?

**Comment by taikitanaka3** [comment]
The document is going to look like this.
image

**Comment by taikitanaka3** [comment]
@rej55 
I updated documents with Execute and Wait button. It will be more appropriate expression

**Comment by rej55** [comment]
I tried this tool.
It is okay for modules included in behavior_velocity_planner.

I think execute button for following modules should be separated from other modules.
Furthermore, it would be better to have separate buttons for individual ids.
- Lane change
- Avoidance
- Pull over
- Pull out

**Comment by taikitanaka3** [comment]
@rej55 
How about this layout? 
|                                |         |      |
| ------------------------------ | ------- | ---- |
| All cooperate command          | Execute | Wait |
| Lateral cooperate command      | Execute | Wait |
| Longitudinal cooperate command | Execute | Wait |

> it would be better to have separate buttons for individual ids
yes I implemented it but it sometimes doesn't work as expected, maybe because of latency of button signal and fast update of rtc status. if it works fine I will add.

**Comment by rej55** [comment]
Longitudinal command is okay but lateral modules should not be placed all execute command.
This is because most lateral modules cannot be canceled once executed.

**Comment by rej55** [comment]
> I tried this tool. It is okay for modules included in behavior_velocity_planner.
> 
> I think execute button for following modules should be separated from other modules. Furthermore, it would be better to have separate buttons for individual ids.
> 
> * Lane change
> * Avoidance
> * Pull over
> * Pull out

This suggestion will be handled in other PR.

**Review by rej55 — APPROVED** [review]
LGTM

**Pull request merged**